envoy/config/xds_manager.h

```diff
@@ -22,6 +23,20 @@ class XdsManager {
 public:
   virtual ~XdsManager() = default;
 
+  /**
+   * Initializes the xDS-Manager.
+   * This should be called after the cluster-manager is created.
+   * @param cm - a pointer to a valid cluster manager.
+   * @return Ok if the initialization was successful, or an error otherwise.
+   */
+  virtual absl::Status initialize(Upstream::ClusterManager* cm) PURE;
```

Comment:
should the param be a reference type instead? E.g. this way the api would imply you can only use it with a valid cm instance

Reply:
I'm open to passing a reference instead of pointer (if it makes sense).
Nevertheless, the XdsManagerImpl class will need to hold a pointer to the cluster-manager and not a reference, as the xds-manager is instantiated before the cluster-manager.
LMK if you still think we should have this argument to a reference, and if so, I'll make the change.

Reply:
in that case everything looks good, i was under impression that cm is constructed before the xds manager.